morgoth/base/static/js/utils/apiFetch.js

```diff
@@ -0,0 +1,70 @@
+const API_ROOT = '/api/v1/';
+
+export default function apiFetch(url, options = {}) {
+  let dispatch;
+  let handleSuccess;
+  let handleError;
+
+  const headers = new Headers();
+  headers.append('Content-Type', 'application/json');
+
+  const settings = {
+    headers,
+    ...options,
+  };
+
+  // Convert `data` to `body` if necessary.
+  if ('data' in settings) {
+    if ('body' in settings) {
+      throw new Error('Only pass one of `settings.data` and `settings.body`.');
+    }
+
+    const method = (settings.method || 'get').toLowerCase();
+
+    if (method !== 'get' && method !== 'head') {
+      settings.body = JSON.stringify(settings.data);
+    }
+
+    delete settings.data;
+  }
+
+  // Extract the `dispatch` function from settings.
+  if (typeof settings.dispatch === 'function') {
+    dispatch = settings.dispatch;
+    delete settings.dispatch;
+  } else {
+    throw new Error('A valid `dispatch` function must be provided.');
+  }
+
+  // Extract the `handleSuccess` function from settings.
+  if (typeof settings.success === 'function') {
+    handleSuccess = settings.success;
+    delete settings.success;
+  } else {
+    throw new Error('A valid `success` function must be provided.');
+  }
+
+  // Extract the `handleError` function from settings.
+  if (typeof settings.error === 'function') {
+    handleError = settings.error;
+    delete settings.error;
+  } else {
+    throw new Error('A valid `error` function must be provided.');
+  }
```

Comment:
Have you considered abusing proptypes for this sort of validation? (Though it isn't actually very good at it...). Either way, it seems like this is something that could be improved.